### ISSUE TITLE
macro expansion takes place in the namespace where `make!` is called

### ISSUE THREAD

**Issue by daslu** [scicloj/clay#132]
Assume we define a namespace `dummy` as follows:

```clj
(ns dummy)
(def x 9)
(defmacro m [] `x)
(m)
```
and that we try to `make!` if from within another namespace, say `user`:
```clj
(require '[scicloj.clay.v2.api :as clay])
(clay/make! {:source-path ["notebooks/dummy.clj"]})
```

Then we receive an error of the following kind:
```clj
Syntax error compiling at ... 
No such var: user/x
```

As explained by @holyjak, the problem is that macro expansion takes place in the calling namespace.

Here is a way to reproduce this problem without any of Clay involved:
```clj
$ clj
Clojure 1.11.2
user=> 
user=> 
(do (eval '(ns dummy))
    (eval '(def x 9))
    (eval '(defmacro m [] `x))
    (eval '(m)))
Syntax error compiling at (REPL:4:12).
No such var: user/x
```

We need to find a way to avoid this problem in Clay.

**Comment by daslu** [comment]
Zulip discussion: https://clojurians.zulipchat.com/#narrow/stream/422115-clay-dev/topic/macro.20expansion.20takes.20place.20in.20the.20ns.20calling.20make!

**Comment by holyjak** [comment]
I am not sure the description is 100%. In my case, I call `make!` in the ns where the macro is defined and used. It looks more like "macro expansion takes place in the namespace of the first soruce-paths file". I.e. I do

```clj
(ns for-scientists.cavity-physics ...)
(defn TeX ...)
(defmacro TeX-> ... `TeX  ...)

(comment
   (clay/make! (assoc {:format [:quarto :html]
                          :base-source-path "notebooks"
                          :source-path ["for_scientists/index.clj" ; <-- remove this one to "fix" the problem
                                        "for_scientists/cavity_physics.clj"]
                          :base-target-path "docs"
                          ;; Turn this to true if you wish to
                          ;; overwrite the whole target dir:
                          :clean-up-target-dir false}
                    :clean-up-target-dir true
                    :show false)))

**Comment by daslu** [comment]
Fixed by #133.
